lib/battle/AccessibilityInfo.cpp

```diff
@@ -25,15 +25,15 @@ bool AccessibilityInfo::accessible(BattleHex tile, bool doubleWide, ui8 side) co
 	//do not use getHexes for speed reasons
 	if(!tile.isValid())
 		return false;
-	if(at(tile) != EAccessibility::ACCESSIBLE && !(at(tile) == EAccessibility::GATE && side == BattleSide::DEFENDER))
+	if(at(tile) != EAccessibility::ACCESSIBLE && (at(tile) != EAccessibility::GATE || side != BattleSide::DEFENDER))
 		return false;
 
 	if(doubleWide)
 	{
 		auto otherHex = battle::Unit::occupiedHex(tile, doubleWide, side);
 		if(!otherHex.isValid())
 			return false;
-		if(at(otherHex) != EAccessibility::ACCESSIBLE && !(at(otherHex) == EAccessibility::GATE && side == BattleSide::DEFENDER))
+		if(at(otherHex) != EAccessibility::ACCESSIBLE && (at(otherHex) != EAccessibility::GATE || side != BattleSide::DEFENDER))

```

Comment:
Any chance we can refactor this code duplication into a method?

Reply:
Also such complex conditions may be better to split onto s few operators. In a separate method it may be way easier to organize

Reply:
Done in separate method.